### PR TITLE
 initial commit to add .net framework to github actions test

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: .github/workflows/dotnetcore.yml [hunk 1]
```diff
@@ -40,16 +41,34 @@ jobs:
       uses: actions/setup-dotnet@v2.1.0
       with:
         dotnet-version: 8.0.x
+
+    - name: 'Build solution'
```
Comment (jmprieur) [inline .github/workflows/dotnetcore.yml:45]: Setup MsBuild?

File: .github/workflows/dotnetcore.yml [hunk 2]
```diff
@@ -40,16 +41,34 @@ jobs:
...
     
     - name: Build with .NET 8.0.x
-      run: dotnet test Microsoft.Identity.Web.sln -f net8.0 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
+      run: dotnet test --no-restore --no-build Microsoft.Identity.Web.sln -f net8.0 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
+
+    - name: Build with .NET 462
+      run: dotnet test --no-restore --no-build Microsoft.Identity.Web.sln -f net462 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
+
+    - name: Build with .NET 472
+      run: dotnet test Microsoft.Identity.Web.sln -f net472 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
```
Comment (jmprieur) [inline .github/workflows/dotnetcore.yml:70]:
```suggestion
      run: dotnet test Microsoft.Identity.Web.sln --no-restore --no-build -f net472 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
```

File: .github/workflows/dotnetcore.yml [hunk 3]
```diff
@@ -40,16 +41,34 @@ jobs:
...
     - name: Build with .NET 6.0.x
-      run: dotnet test Microsoft.Identity.Web.sln -f net6.0 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName!~IntegrationTests)"
+      run: dotnet test --no-restore --no-build Microsoft.Identity.Web.sln -f net6.0 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName!~IntegrationTests)"
 
     - name: Build with .NET 7.0.x
-      run: dotnet test Microsoft.Identity.Web.sln -f net7.0 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
+      run: dotnet test --no-restore --no-build Microsoft.Identity.Web.sln -f net7.0 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
     
     - name: Build with .NET 8.0.x
```
Comment (jmprieur) [inline .github/workflows/dotnetcore.yml:63]:
```suggestion
    - name: Test with .NET 8.0.x
```

File: .github/workflows/dotnetcore.yml [hunk 4]
```diff
@@ -40,16 +41,34 @@ jobs:
...
-      run: dotnet test Microsoft.Identity.Web.sln -f net8.0 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
+      run: dotnet test --no-restore --no-build Microsoft.Identity.Web.sln -f net8.0 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
+
+    - name: Build with .NET 462
```
Comment (jmprieur) [inline .github/workflows/dotnetcore.yml:66]:
```suggestion
    - name: Test with .NET 462
```

File: .github/workflows/dotnetcore.yml [hunk 5]
```diff
@@ -40,16 +41,34 @@ jobs:
...
+      run: dotnet test --no-restore --no-build Microsoft.Identity.Web.sln -f net462 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
+
+    - name: Build with .NET 472
```
Comment (jmprieur) [inline .github/workflows/dotnetcore.yml:69]:
```suggestion
    - name: Test with .NET 472
```

File: .github/workflows/dotnetcore.yml [hunk 6]
```diff
@@ -40,16 +41,34 @@ jobs:
...
+      run: dotnet test Microsoft.Identity.Web.sln -f net472 -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName=IntegrationTests)"
+
+    - name: Build with .NET standard2.0
```
Comment (jmprieur) [inline .github/workflows/dotnetcore.yml:72]:
```suggestion
    - name: Test with .NET standard2.0
```

**Review by jmprieur — APPROVED** [review]
LGTM
I added a few naming suggestions for the tasks.

I wonder if the following (without precising any `-f` wouldn't just test all frameworks)?

```
run: dotnet test --no-restore --no-build Microsoft.Identity.Web.sln -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName!~IntegrationTests)"
```

File: .github/workflows/dotnetcore.yml (hunk 1 above)
Comment (jennyf19) [reply .github/workflows/dotnetcore.yml:45]: done.

**Comment by jennyf19** [comment]
> LGTM I added a few naming suggestions for the tasks.
> 
> I wonder if the following (without precising any `-f` wouldn't just test all frameworks)?
> 
> ```
> run: dotnet test --no-restore --no-build Microsoft.Identity.Web.sln -p:FROM_GITHUB_ACTION=true --configuration Release --filter "(FullyQualifiedName!~Microsoft.Identity.Web.Test.Integration)&(FullyQualifiedName!~WebAppUiTests)&(FullyQualifiedName!~IntegrationTests)"
> ```

Does not work, as the -p is not respected so tests which cannot run via GitHub action are being run. Otherwise, would have been awesome

File: .github/workflows/dotnetcore.yml (hunk 3 above)
Comment (jennyf19) [reply .github/workflows/dotnetcore.yml]: fixed

File: .github/workflows/dotnetcore.yml (hunk 2 above)
Comment (jennyf19) [reply .github/workflows/dotnetcore.yml:69]: fixed

**Pull request merged**